org-roam.el

```diff
@@ -587,6 +591,42 @@ This needs to be quick/infrequent, because this is run at
     (call-process org-roam-graphviz-executable nil 0 nil temp-dot "-Tsvg" "-o" temp-graph)
     (call-process org-roam-graph-viewer nil 0 nil temp-graph)))
 
+(defun org-roam--rename-file-links (file new-file &rest args)
+  "Rename backlinks of FILE to refer to NEW-FILE."
+  (when (and (not (auto-save-file-name-p file))
+             (not (auto-save-file-name-p new-file))
+             (org-roam--org-roam-file-p new-file))
+    (org-roam--ensure-cache-built)
+
+    (let* ((files (gethash file org-roam-backward-links-cache nil))
+           (path (org-roam--relative-path file))
+           (new-path (org-roam--relative-path new-file))
+
+           (slug (org-roam--get-title-or-slug file))
+           (old-title (format org-roam-link-title-format slug))
+
+           (new-slug (or (org-roam--get-title-from-cache (file-truename file))
+                         (org-roam--get-title-or-slug new-path)))
+           (new-title (format org-roam-link-title-format new-slug)))
+      (when files
+        (save-excursion
+          (maphash (lambda (file-from contents)
+                     (let* ((relative-path (file-relative-name new-path (file-name-directory file-from))))
+                       (with-temp-file file-from
+                         (insert-file-contents file-from)
```

Comment:
Rather then using a temp file, we can use `find-file-noselect` which will prevent us from needing to reopen buffers we already have open.

Reply:
Do we not have more control over secondary effects using `with-temp-file`? One of which is actually displaying the buffer. As I see it, it would be ideal if we would never opened a file.

Reply:
`find-file-noselect` will not display the buffer, it just returns the buffer. 

If we don't want to open extra files we could either check if it is open already with `get-file-buffer` or use `with-temp-buffer`. But we don't have a need for a temporary file. 

```elisp
(with-temp-buffer
  (insert-file-contents file-from)
  ;; do stuff
  (write-file file-from))
```

Reply:
That would be ideal. However, I think some functionality relies on at least the buffer name being associated with a file, so opening the file might be unavoidable if we want to update the cache.